### ISSUE TITLE
Getting text/audio embeddings (and their gradients) from the pretrained models. 

### ISSUE THREAD

**Issue by hugofloresgarcia** [zhaoyanpeng/lvamodel#1]
Hi! 

First of all, awesome work!
I'd like to be able to load a pretrained model so I can obtain text and audio embeddings (and their gradients). Could you help me figure out how to do that?

Thanks! :)

**Comment by hugofloresgarcia** [comment]
Also, I noticed that the audio preprocessing is done within kaldi, and not with torch preprocessing. Would that mean that I wouldn't be able to run gradients through the mel spectrogram transform?

**Comment by zhaoyanpeng** [comment]
hi! I am using [kaldi APIs](https://github.com/zhaoyanpeng/vipant/blob/93b06ff43ae6a76323cecea4c10cf457945c2711/cvap/data/audio/transform.py#L29) of [torchaudio](https://pytorch.org/audio/stable/compliance.kaldi.html#fbank). I think you are right: the transform function does not seem to produce any gradients, so no way to run gradients through fbank / spectrogram transformation.


 If you want to get the gradients of audio embeddings output by the audio encoder, maybe try [torch.autograd.grad](https://pytorch.org/docs/stable/generated/torch.autograd.grad.html#torch-autograd-grad).

**Comment by zhaoyanpeng** [comment]
if you want to encode your own audio-text data w/ a pre-trained VA model, you would need to modify this [function](https://github.com/zhaoyanpeng/vipant/blob/93b06ff43ae6a76323cecea4c10cf457945c2711/cvap/model/clap.py#L42-L53) to directly save the audio and text features.

You might find this AT retrieval [script](https://gist.github.com/zhaoyanpeng/3cfaddbc5c7c8c7aeb20e5410bb75930) helpful.

Let me know if that works for you!
